playground/backend/internal/cache/redis/redis_cache.go

```diff
@@ -102,9 +102,12 @@ func unmarshalBySubKey(subKey cache.SubKey, value string) (result interface{}, e
 	case cache.Status:
 		result = new(pb.Status)
 		err = json.Unmarshal([]byte(value), &result)
-	case cache.RunOutput, cache.CompileOutput:
+	case cache.RunOutput, cache.RunError, cache.CompileOutput:
 		result = ""
 		err = json.Unmarshal([]byte(value), &result)
+	case cache.Canceled:
+		result = false
+		err = json.Unmarshal([]byte(value), &result)
```

Comment:
I see that this Unmarshal step happens in each switch case – can we do it after the switch-case structure since it is common?

Reply:
Moved it after switch-case part.